SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/TwinMergeHelper.cs

```diff
@@ -44,6 +44,32 @@ internal static bool TryCreatePatchDocument(BasicDigitalTwin existingDigitalTwin
             return !jsonPatchDocument.IsEmpty();
         }
 
+        internal static bool TryCreatePatchDocument(BasicRelationship existingRelationship, BasicRelationship newRelationship, out JsonPatchDocument jsonPatchDocument)
+        {
+            jsonPatchDocument = new JsonPatchDocument();
+
+            foreach (var newProperty in newRelationship.Properties)
+            {
+                var newPropertyName = newProperty.Key;
+                var newPropertyValue = newProperty.Value;
+
+                // See if the existing relationship has the property
+                if (existingRelationship.Properties.TryGetValue(newPropertyName, out var existingPropertyValue))
+                {
+                    if (existingPropertyValue != null && existingPropertyValue != newPropertyValue)
+                    {
+                        jsonPatchDocument.AppendReplace("/" + newPropertyName, newPropertyValue);
```

Comment:
Is there any AppendRemove that needs to be added as DTDL models evolve? Seems scary to remove, but wondering how DTDL validation will behave as we run through this

Reply:
I don't think we ever want to remove models, as they may not have been created by this process, and the last thing we want to do is delete data manually entered or added out of process.